php/containers.json

```
@@ -612,7 +612,7 @@
       "internal_port": "9200",
       "environment": [
         "TZ=%TIMEZONE%",
-        "ES_JAVA_OPTS=-Xms512M -Xmx512M",
+        "ES_JAVA_OPTS=-Xms512M -Xmx1024M",
```

I would be fine with this. Wdyt @Zoey2936?
```suggestion
        "ES_JAVA_OPTS=-Xms1024M -Xmx1024M",
```

---

we can do this, but it will use a lot of ram

---

@szaimen I don't understand why you would rewrite this with the same value for initial and max heap size, I've always seen this with Xms < Xmx.  
I'm almost sure this is the source of the problem, when java cannot allocate more than initial (reserved) allocated size, it seems elasticsearch cannot spawn new nodes for indexing.  

I could try `-Xms256M -Xmx512M` but it's very long to re-index...
I have a lot of files and groupfolders, but also ignored dirs, yet I'm pretty sure I don't need 1GB or 2GB of RAM to do this. I don't know about the memory usage printed by occ, but I've never seen it exceed 200 or 300MB.

---

> we can do this, but it will use a lot of ram

To avoid high memory usage even if not that much is required, shouldn't we use a smaller value like `-Xms64m -Xmx1024M` ?

---

I have another idea : may be I should close this, in favor of another issue for feature request : for a `ES_JAVA_OPTS` env that can be set in the compose file ?  

In my case the problem is probably also related to groupfolders, which fulltextsearch does not seems understand really well and I wouldn't be surprised if files are indexed as many times as they are users. It's not clears from the logs if there's just an entry per user (file path) or if file is actually parsed every time.  
But I don't think this high memory usage is normal in regard to the number of files I have to index here.

---

Hm... 512m and 512m are the values recommended in https://github.com/R0Wi/elasticsearch-nextcloud-docker#how-to-use-this so not sure increasing the value is the way to go. However I would really like to not introduce a variable here. So maybe the fulltextsearch app needs to be improved instead in order to improve the handling of groupfolders instead?

---

As you want, yet if I were you I wouldn't take anything for granted except if it comes from the official ElasticSearch docs.  

But if this does not happen for users without groupfolders, I agree here isn't the place to make that change.  

Yet I don't understand why want to avoid a variable, since this setting has to be set according to number of files to scan, you might see this exception pop in future issues !  

At least it should be added to the docs which file has to be modified (the `containers.json` in order for the first indexing to succeed. This is not a problem if the container then re-run with Xmx512M, because it's only happening during a first run (or manual `occ fulltextsearch:index`.

---

Well I just gave a quick look at ES docs, they show the following example for memory settings in docker. So yeah I guess my first argument is broken ^^

`docker run -e ES_JAVA_OPTS="-Xms1g -Xmx1g" -e ENROLLMENT_TOKEN="<token>" --name es01 -p 9200:9200 --net elastic -it docker.elastic.co/elasticsearch/elasticsearch:8.9.2`

---

But interestingly, the following is stated, just under this example : 
>The ES_JAVA_OPTS variable overrides all other JVM options. We do not recommend using ES_JAVA_OPTS in production.

---

running java without xms/xmx will make java to eat basically all ram

---

Yes, I'm confused by this statement, if we run in docker, isn't it for ES java process to be alone in its env ? So I don't see how ES_JAVA_OPTS could override other JVM options.. how many JVM processes are there in this container 😬